### PR TITLE
`Paywalls`: set up CI tests and API Tester

### PR DESCRIPTION
**Pull request by NachoSoto**
### Description:
The new `spm-revenuecat-ui-ios-16` CircleCI job does the following:
- Build using `SPM` on release configuration, on `macOS` (because SPM on command line can only build for the target architecture)
- Run tests on iOS 16 simulator
- Compile new API tests

### Other changes:
- Fixed a couple of warnings building `RevenueCatUI`
- Fixed `macOS` `RevenueCatUI` build (even though it will be disabled in #2821).

### CODE REVIEW

File: Tests/APITesters/RevenueCatUIAPITester/SwiftAPITester/PaywallViewAPI.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import SwiftUI
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, *)
+struct App: View {
+
+    private var offering: Offering
+    private var paywall: PaywallData
+
+    var body: some View {
+        PaywallView(offering: self.offering, paywall: self.paywall)
```
Comment (NachoSoto) [inline Tests/APITesters/RevenueCatUIAPITester/SwiftAPITester/PaywallViewAPI.swift:19]: Literally the only API right now :)

**Comment by NachoSoto** [comment]
Not sure why API tests aren't building in CI, I'll need to look into it.

File: Tests/APITesters/RevenueCatUIAPITester/SwiftAPITester/main.swift
Comment (vegaro) [inline Tests/APITesters/RevenueCatUIAPITester/SwiftAPITester/main.swift:1]: Is this file needed?

File: fastlane/Fastfile [hunk 2]
```diff
@@ -492,6 +513,8 @@ platform :ios do
       './Tests/InstallationTests/SPMCustomEntitlementComputationInstallation/SPMInstallation.xcodeproj/project.pbxproj',
       './Tests/InstallationTests/ReceiptParserInstallation/ReceiptParserInstallation.xcodeproj/project.pbxproj',
       './Tests/APITesters/CustomEntitlementComputationSwiftAPITester/CustomEntitlementComputationSwiftAPITester.xcodeproj/project.pbxproj',
+      # Fixme: this project is currently pointing to `paywalls` branch, it needs to be changed to `main` once that's merged.
```
Comment (vegaro) [inline fastlane/Fastfile:516]: do you need to update line 527?

Comment (NachoSoto) [reply fastlane/Fastfile:516]: Oh I already did this so let me remove the comment thanks :D

Comment (NachoSoto) [reply fastlane/Fastfile]: The project is pointing to `main` so this works now.

File: Tests/APITesters/RevenueCatUIAPITester/SwiftAPITester/main.swift
Comment (NachoSoto) [reply Tests/APITesters/RevenueCatUIAPITester/SwiftAPITester/main.swift:1]: Yeah the target needs a `@main` function.

**Review by tonidero — APPROVED** [review]
Changes look good to me. Left a comment but can be done separately.

File: .circleci/config.yml [hunk 3]
```diff
@@ -314,6 +314,25 @@ jobs:
           command: swift build -c release --target ReceiptParser
           no_output_timeout: 30m
 
+  spm-revenuecat-ui-ios-16:
```
[the remark this answers is not in the document]
Comment (NachoSoto) [reply .circleci/config.yml:317]: That’s what we came up with. @RevenueCat/cashnip any thoughts on `RevenueCatPaywallsUI`?
